app/javascript/components/server-components/Discover/index.tsx

```diff
@@ -370,29 +420,46 @@ const Discover = (props: Props) => {
               sort: state.params.query || hasOfferCode ? "default" : state.params.sort,
             }}
             appendFilters={
-              <details>
-                <summary>Rating</summary>
-                <fieldset role="group">
-                  {range(4, 0).map((number) => (
-                    <label key={number}>
-                      <span className="flex shrink-0 items-center gap-1">
-                        <RatingStars rating={number} />
-                        and up
-                      </span>
-                      <input
-                        type="radio"
-                        value={number}
-                        aria-label={`${number} ${number === 1 ? "star" : "stars"} and up`}
-                        checked={number === state.params.rating}
-                        readOnly
-                        onClick={() =>
-                          updateParams(state.params.rating === number ? { rating: undefined } : { rating: number })
-                        }
-                      />
-                    </label>
-                  ))}
-                </fieldset>
-              </details>
+              <>
+                <details>
+                  <summary>Rating</summary>
+                  <fieldset role="group">
+                    {range(4, 0).map((number) => (
+                      <label key={number}>
+                        <span className="flex shrink-0 items-center gap-1">
+                          <RatingStars rating={number} />
+                          and up
+                        </span>
+                        <input
+                          type="radio"
+                          value={number}
+                          aria-label={`${number} ${number === 1 ? "star" : "stars"} and up`}
+                          checked={number === state.params.rating}
+                          readOnly
+                          onClick={() =>
+                            updateParams(state.params.rating === number ? { rating: undefined } : { rating: number })
+                          }
+                        />
+                      </label>
+                    ))}
+                  </fieldset>
+                </details>
+                {hasOfferCode ? (
```

Comment:
Added this:
- Clicking on "Clear" removes the offer code
- Clicking on the X button removes the offer code
<img width="426" height="370" alt="Image" src="https://github.com/user-attachments/assets/8998318e-f7b8-458f-81b7-d7bdcdb859ee" />